Fix #5214, Fix Part of #1723 : Fix onboarding issue faced by new contributors (#5210)
oppia/oppia-android: wiki/Installing-Oppia-Android.md

   
   **Direct download Url**: [Windows](https://redirector.gvt1.com/edgedl/android/studio/install/2021.1.1.23/android-studio-2021.1.1.23-windows.exe) | [Linux](https://redirector.gvt1.com/edgedl/android/studio/ide-zips/2021.1.1.23/android-studio-2021.1.1.23-linux.tar.gz) | [Intel Mac](https://redirector.gvt1.com/edgedl/android/studio/install/2021.1.1.23/android-studio-2021.1.1.23-mac.dmg) | [Apple Silicon Mac](https://redirector.gvt1.com/edgedl/android/studio/install/2021.1.1.23/android-studio-2021.1.1.23-mac_arm.dmg)

   You can have multiple versions of Android Studio installed on your machine at the same time.

2. Configure your Android Studio
   - In Android Studio, open Tools > SDK Manager.
     - In the "SDK Platforms" tab (which is the default), select `API Level 28` and also `API Level 30` (for Bazel support).
     - Also, navigate to the "SDK Tools" tab, click the "Show Package Details" checkbox at the bottom right, then click on "Android SDK Build-Tools 34-rc1" and select 29.0.2 (this is needed for Bazel support).

   - Then, click "Apply" to download and install these two SDKs/Tools.
   

>    - Must have **JDK 11** is selected:

Reviewer: ```suggestion
   - Must have **JDK 11** selected:
```

Author: Done.

## Set up and run tests
Testing the app is an integral part of our development process. You will need to test all code changes to ensure that the app works correctly, therefore it is important to ensure that your test configuration works.

> Robolectric allows for faster unit testing of Android applications without the need for device or emulator setup.

Reviewer: ```suggestion
We strongly recommend running tests on Robolectric which is faster because it does not require a physical device or emulator setup.
```

Author: Done.

> ### Configure Robolectric Tests

Reviewer: @MohitGupta121, these instructions were originally written for only non-app module tests, but we also now need to include instructions to run app module tests on gradle(the Junit configuration). Please include a step by step guide here to run the tests such as SplashActivityTest on Robolectic, and ensure that these are also updated on the dedicated testing page as well.
<img width="1386" alt="Screenshot 2023-10-28 at 10 41 43" src="https://github.com/oppia/oppia-android/assets/59600948/6164bae2-817b-4f17-a045-45fa87d8087a">

Author: Done.

Reviewer: This looks great now, thanks.

> These are tests that are in non-app modules, such as **domain** or **utility**.

Reviewer: This needs to be modified now.

Author: Done.

Reviewer: I believe this should be a subheading the same way you had "Step-by-Step guidance for setting up and running app modules robolectric test:" above, you can also make this bold, and someting like 
"Step-by-Step guidance for setting up and running non-app modules robolectric test:
These are tests that are in other  modules, such as **domain** or **utility**.

Also add some spacing so it is clear that it is a different section.

Reviewer: I think you may have misunderstood what I meant here, I have added a suggestion as to what I meant.

Reviewer: Basically, I needed it to be clear that we usually have different modules and that the test initialization will be different. The instructions are correct, but the section subtitle was not clear for non-app module tests.

1. In Android Studio, open the desired test file, e.g., `AnalyticsControllerTest`.
2. In the test file, to the left of the class name, click on the orange and green arrow, and select **Run 'AnalyticsControllerTest'**.
   - You will notice that the emulator is greyed out, but the run window will open to show the running tests:
   - ![](https://user-images.githubusercontent.com/59600948/272657015-158117e5-47d2-40fc-a38b-5dee6c347556.png)

You can find step-by-step guidance for setting up and running Robolectric Tests in this [wiki](https://github.com/oppia/oppia-android/wiki/Oppia-Android-Testing#robolectric).

> ***Espresso is slower for unit testing Android applications compared to Robolectric, so we recommend to use Robolectric.***

Reviewer: ```suggestion
**Espresso is slower for running tests, so we recommend using Robolectric.**
```

Author: Done.

3. Enter following information:
   - a) Name of test. Example: In my case "SplashActivityTest"
   - b) Make sure select "java 11" and oppia-android.app
   - c) Class path of Test class. Example: In my case "org.oppia.android.app.splash.
   SplashActivityTest"
   - d) Press `OK` to select the test.
   ![](https://github.com/oppia/oppia-android/assets/76530270/5901624a-df76-4b27-8f31-6077a68fcb89)

4. Click on "Run" button to run robolectric test. (In my case "SplashActivityTest")
   ![](https://github.com/oppia/oppia-android/assets/76530270/75a6b998-90c5-4f0a-8886-78f96970be90)

These are tests that are in non-app modules, such as **domain** or **utility**.
1. In Android Studio, open the desired test file, e.g., `AnalyticsControllerTest`.
2. In the test file, to the left of the class name, click on the orange and green arrow, and select **Run 'AnalyticsControllerTest'**.
   - You will notice that the emulator is greyed out, but the run window will open to show the running tests:
   - ![](https://user-images.githubusercontent.com/59600948/272657015-158117e5-47d2-40fc-a38b-5dee6c347556.png)

> **Espresso is slower for running tests, so we recommend using Robolectric.**

Reviewer: I think this needs to go to the next line right below "### Configure Emulator Tests"

Author: Done.

We strongly recommend running tests on Robolectric which is faster because it does not require a physical device or emulator setup.

### Configure Robolectric Tests

#### Step-by-Step guidance for setting up and running app modules robolectric test:

1. Go to **Edit Configuration** in Android Studio (Bumblebee | 2021.1.1 Patch 3)
   ![](https://user-images.githubusercontent.com/9396084/79109714-83525980-7d96-11ea-99d7-f83ea81a8a50.png)

2. Click on Add(+) -> **JUnit**
   ![](https://github.com/oppia/oppia-android/assets/76530270/87caf3fc-37d9-472d-92fd-b8ec49fb6b49)

3. Enter following information:
   - a) Name of test. Example: In my case "SplashActivityTest"
   - b) Make sure select "java 11" and oppia-android.app
   - c) Class path of Test class. Example: In my case "org.oppia.android.app.splash.
   SplashActivityTest"
   - d) Press `OK` to select the test.
   ![](https://github.com/oppia/oppia-android/assets/76530270/5901624a-df76-4b27-8f31-6077a68fcb89)

4. Click on "Run" button to run robolectric test. (In my case "SplashActivityTest")
   ![](https://github.com/oppia/oppia-android/assets/76530270/75a6b998-90c5-4f0a-8886-78f96970be90)

> #### These are tests that are in non-app modules, such as **domain** or **utility**:

Reviewer: ```suggestion
#### For tests that are in non-app modules, such as **domain** or **utility**:
```

Author: Done.
